src/main/java/htsjdk/samtools/CRAMBAIIndexer.java

```diff
@@ -100,17 +101,19 @@ public CRAMBAIIndexer(final File output, final SAMFileHeader fileHeader) {
      * @param fileHeader header for the corresponding bam file.
      */
     public CRAMBAIIndexer(final OutputStream output, final SAMFileHeader fileHeader) {
-
+        if (fileHeader.getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new SAMException("CRAM file must be coordinate-sorted for indexing.");
+        }
         numReferences = fileHeader.getSequenceDictionary().size();
         indexBuilder = new BAMIndexBuilder(fileHeader);
         outputWriter = new BinaryBAMIndexWriter(numReferences, output);
     }
 
     /**
-     * Index a container, any of mapped, unmapped and multiple references are allowed. The only requirement is sort
-     * order by coordinate.
-     * For multiref containers the method reads the container through unpacking all reads. This is slower than single
-     * reference but should be faster than normal reading.
+     * Index a container, any of mapped, unmapped and multiple references are allowed.
+     * The only requirement is sort order by coordinate.
+     * For multiref containers the method reads the container through unpacking all reads.
+     * This is slower than single reference but should be faster than normal reading.

```

Comment:
No idea what this means.  It was there already.

Reply:
It's not possible to generate a BAI index from CRAI entries (#531) but I'll look into some consolidation.